cmd/metal-api/internal/datastore/machine.go

```diff
@@ -499,6 +503,43 @@ func (rs *RethinkStore) FindWaitingMachine(projectid, partitionid, sizeid string
 	return &newMachine, nil
 }
 
+// checkSizeReservations returns true when an allocation is possible and
+// false when size reservations prevent the allocation for the given project in the given partition
+func checkSizeReservations(available metal.Machines, projectid, partitionid string, machinesByProject map[string]metal.Machines, size metal.Size) bool {
+	if size.Reservations == nil {
+		return true
+	}
+
+	var (
+		reservations = 0
+		max          = func(x, y int) int {
```

Comment:
There is already min/max available in go since >= 1.21, see: https://tip.golang.org/ref/spec#Min_and_max